pkg/api/v1alpha2/types_config.go

```diff
@@ -0,0 +1,169 @@
+// Setting MaxVersion will override this setting.
+func (r ReleaseChannel) IsHeadsOnly() bool {
+	return !r.Full
+}
+
+// Operator defines the configuration for operator catalog mirroring.
+type Operator struct {
+	// Mirror specific operator packages, channels, and versions, and their dependencies.
+	// If HeadsOnly is true, these objects are mirrored on top of heads of all channels.
+	// Otherwise, only these specific objects are mirrored.
+	IncludeConfig `json:",inline"`
+
+	// Catalog image to mirror. This image must be pullable and available for subsequent
+	// pulls on later mirrors.
+	// This image should be an exact image pin (registry/namespace/name@sha256:<hash>)
+	// but is not required to be.
+	Catalog string `json:"catalog"`
+	// Full defines whether all packages within the catalog
+	// or specified IncludeConfig will be mirrored or just channel heads.
+	Full bool `json:"full,omitempty"`
+	// SkipDependencies will not include dependencies
+	// of bundles included in the diff if true.
+	SkipDependencies bool `json:"skipDeps,omitempty"`
+}
+
+// IsHeadsOnly determine if the mode set mirrors only channel heads of all packages in the catalog.
+// Channels specified in DiffIncludeConfig will override this setting;
+// heads will still be included, but prior versions may also be included.
+func (o Operator) IsHeadsOnly() bool {
+	return !o.Full
+}
+
+// Helm defines the configuration for Helm chart download
+// and image mirroring
+type Helm struct {
+	// Repo is the helm repository containing the charts
+	Repos []Repo `json:"repos,omitempty"`
+	// Local is the configuration for locally stored helm charts
+	Local []Chart `json:"local,omitempty"`
+}
+
+// Repo defines the configuration for a Helm repo.
+type Repo struct {
+	// URL is the url of the Helm repository
+	URL string `json:"url"`
+	// Name is the name of the Helm repository
+	Name string `json:"name"`
+	// Charts is a list of charts to pull from the repo
+	Charts []Chart `json:"charts"`
+}
+
+// Chart is the information an individual Helm chart
+type Chart struct {
+	// Chart is the chart name as define
+	// in the Chart.yaml or in the Helm repo.
+	Name string `json:"name"`
+	// Version is the chart version as define in the
+	// Chart.yaml or in the Helm repo.
+	Version string `json:"version,omitempty"`
+	// Path defines the path on disk where the
+	// chart is stored.
+	// This is applicable for a local chart.
+	Path string `json:"path,omitempty"`
+	// ImagePaths are custom JSON paths for images location
+	// in the helm manifest or templates
+	ImagePaths []string `json:"imagepaths,omitempty"`
+}
+
+// Image contains image pull information.
+type Image struct {
+	// Name of the image. This should be an exact image pin (registry/namespace/name@sha256:<hash>)
+	// but is not required to be.
+	Name string `json:"name"`
+}
+
+// AdditionalImages defines the configuration
+// individual image content types.
+type AdditionalImages struct {
```

Comment:
Each of the following 3 structures is identical, how about having a single structure, `Images` for example, and then in `Mirror` struct you'd just use:
```
	AdditionalImages []Images `json:"additionalImages,omitempty"`
	BlockedImages []Images `json:"blockedImages,omitempty"`
...
```

unless you can explain me the difference that you anticipate between these 3 structures.

Reply:
Less structures make the API cleaner and more readable :wink:

Reply:
Ack. Will incorporate it into a follow on PR.